storage_service/locations/models/package.py

```diff
@@ -894,6 +900,27 @@ def start_reingest(self, pipeline, reingest_type):
         elif reingest_type == self.METADATA_ONLY:
             reingest_files.append(os.path.join(relative_path, 'data', 'objects', 'metadata', ''))
 
+        # Fetch processing configuration, put it in the root of the package and
+        # include the file in reingest_files.
+        if reingest_type == self.FULL and processing_config != 'default':
+            try:
+                config = pipeline.get_processing_config(processing_config)
+            except PipelineClientException:
+                LOGGER.exception('Reingest: processing configuration %s could not be loaded, error: %s', processing_config, e)
+                pass
+            else:
+                config_path = os.path.join(local_path, 'processingMCP.xml')
+                try:
+                    # It's not expected to find an existing processingMCP.xml
+                    # file in the original AIP, but we are using the w+ mode
+                    # just in case.
+                    with open(config_path, 'w+') as f:
+                        f.write(config)
+                    LOGGER.debug('Reingest: processing configuration %s written, location: %s', processing_config, config_path)
+                    reingest_files.append(os.path.join(relative_path, 'processingMCP.xml'))
+                except IOError as e:
+                    LOGGER.exception('Reingest: processing configuration %s could not be written, error: %s', processing_config, e)
```

Comment:
LOGGER.exception will automatically log the exception details including name, so it doesn't need to be written in the error message.

If this error happens, do we want to continue processing, or should this re-raise?